common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/WebViewAuthorizationFragment.java

```diff
@@ -140,6 +148,87 @@ public void onCreate(@Nullable Bundle savedInstanceState) {
         }
     }
 
+    @Override
+    public void onResume() {
+        super.onResume();
+        final String methodTag = TAG + ":onResume";
+        final Bundle extras = getExtras();
+        if (isSwitchBrowserResumeFlow(extras)) {
+            Logger.info(methodTag, "Resume switch browser protocol on WebView.");
+            resumeSwitchBrowser(extras);
+        }
+    }
+
+    /**
+     * Get the extras from the activity intent.
+     *
+     * @return Bundle with the extras
+     */
+    @NonNull
+    private Bundle getExtras() {
+        final Activity activity = getActivity();
+        if (activity == null) {
+            return Bundle.EMPTY;
+        }
+        final Intent intent = activity.getIntent();
+        if (intent == null) {
+            return Bundle.EMPTY;
+        }
+        final Bundle extras =  intent.getExtras();
+        return extras == null ? Bundle.EMPTY : extras;
+    }
+
+    /**
+     * Check if the extras contains the switch browser code and action uri.
+     * also checks that the fragment is waiting for the switch browser resume endpoint.
+     * if so, it means we are resuming the switch browser flow.
+     *
+     * @param extras Bundle
+     * @return boolean
+     */
+    private boolean isSwitchBrowserResumeFlow(@NonNull final Bundle extras) {
+        final String methodTag = TAG + ":isSwitchBrowserResumeFlow";
+        final boolean containsActionUri = extras.containsKey(
```

Comment:
Are we also checking the path to validate?

redirect_uri/switch_browser_resume

I think we should validate that.

If action_uri and code are missing while in a switch browser flow then that should be considered a bug.

Reply:
yes, we are validating the path on the BrokerBrowserRedirectActivity, reworked, so here if action_uri and code are missing and is a resume flow we consider it an error